modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,846 @@
+                        dictionary.value<std::string>(LabelAlignmentOptionInfo.identifier);
+                    if (labelAlignment == "Horizontally") {
+                        _labelAlignmentOption = Horizontally;
+                    }
+                    else {
+                        _labelAlignmentOption = Circularly;
+                    }
+                }
+
+                _font = font;
+                initializeFonts();
+            }
+        }
+    }
+}
+
+bool GlobeLabelsComponent::initializeGL() {
+    return true;
+}
+
+void GlobeLabelsComponent::initializeFonts() {
+    if (_font == nullptr) {
+        _font = openspace::global::fontManager.font(
+            "Mono",
+            static_cast<float>(_labelsFontSize),
+            ghoul::fontrendering::FontManager::Outline::Yes,
+            ghoul::fontrendering::FontManager::LoadGlyphs::No
+        );
+    }
+}
+
+
+bool GlobeLabelsComponent::deinitialize() {
+    return true;
+}
+
+bool GlobeLabelsComponent::isReady() const {
+    return true;
+}
+
+void GlobeLabelsComponent::update() {
+        
+}
+
+bool GlobeLabelsComponent::loadLabelsData(const std::string& file) {
+    bool success = true;
+    if (_labelsDataPresent) {
+        std::string cachedFile = FileSys.cacheManager()->cachedFilename(
+            ghoul::filesystem::File(file),
+            "GlobeLabelsComponent|" + identifier(),
+            ghoul::filesystem::CacheManager::Persistent::Yes
+        );
+
+        bool hasCachedFile = FileSys.fileExists(cachedFile);
+        if (hasCachedFile) {
+            LINFO(fmt::format(
+                "Cached file '{}' used for labels file '{}'",
+                cachedFile,
+                file
+            ));
+
+            success = loadCachedFile(cachedFile);
+            if (success) {
+                return true;
+            }
+            else {
+                FileSys.cacheManager()->removeCacheFile(file);
+                // Intentional fall-through to the 'else' to generate the cache
+                // file for the next run
+            }
+        }
+        else {
+            LINFO(fmt::format("Cache for labels file '{}' not found", file));
+        }
+        LINFO(fmt::format("Loading labels file '{}'", file));
+
+        success = readLabelsFile(file);
+        if (!success) {
+            return false;
+        }
+
+        success &= saveCachedFile(cachedFile);
+    }
+    return success;
+}
+
+bool GlobeLabelsComponent::readLabelsFile(const std::string& file) {
+    try {
+        std::fstream csvLabelFile(file);
+        if (!csvLabelFile.good()) {
+            LERROR(fmt::format("Failed to open labels file '{}'", file));
+            return false;
+        }
+        if (csvLabelFile.is_open()) {
+            char line[4096];
+            _labels.labelsArray.clear();
+            while (!csvLabelFile.eof()) {
+                csvLabelFile.getline(line, 4090);
+                if (strnlen(line, 4090) > 10) {
```

Comment:
Prefer to create a `std::string` from this and then ditch the C functions
 We have a tokenizer in ghoul